firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -35,10 +43,23 @@
 public class FirebaseInstallations implements FirebaseInstallationsApi {
 
   private final FirebaseApp firebaseApp;
+  private final FirebaseInstallationServiceClient serviceClient;
+  private final FiidCache localCache;
+  private final Executor executor;
 
   /** package private constructor. */
   FirebaseInstallations(FirebaseApp firebaseApp) {
+    this(firebaseApp, new FiidCache(firebaseApp), new FirebaseInstallationServiceClient());
+  }
+
+  FirebaseInstallations(
+      FirebaseApp firebaseApp,
+      FiidCache localCache,
+      FirebaseInstallationServiceClient serviceClient) {
     this.firebaseApp = firebaseApp;
+    this.serviceClient = serviceClient;
+    this.executor = Executors.newFixedThreadPool(6);
```

Comment:
Why is this not a constant?
Why 6?

Reply:
Changed it. Its a random number, based on 3 APIs I chose double the number of available threads.